core/auth/src/main/java/software/amazon/awssdk/auth/credentials/ProcessCredentialsProvider.java

```diff
@@ -174,21 +174,10 @@ private Instant credentialExpirationTime(JsonNode credentialsJson) {
     }
 
     /**
-     * Get a textual value from a json object, throwing an exception if the node is missing or not textual.
+     * Get a textual value from a json object.
      */
     private String getText(JsonNode jsonObject, String nodeName) {
-        JsonNode subNode = jsonObject.get(nodeName);
-
-        if (subNode == null) {
-            return null;
-        }
-
-        if (!subNode.isTextual()) {
-            throw new IllegalStateException(nodeName + " from credential process should be textual, but was " +
-                                            subNode.getNodeType());
-        }
-
-        return subNode.asText();
+        return jsonObject.get(nodeName).map(JsonNode::text).orElse(null);
```

Comment:
We validate the result elsewhere, so the invalid response type will be handled elsewhere.

Reply:
This change introduces a bit of a relaxing of the error messages. Before, they could not specify a value of 1 or null. Now they can, but it'll fail when it hits the service.